### PR TITLE
gh #233 dsAudio Enhance python tests

### PR DESCRIPTION
**Pull request by bhanucbp**
- Updated the L3 tests to avoid the repeated code
- Added ds helper class
- Updated dsAudio class to deploy the binaries which are listed in configuration file to target

### CODE REVIEW

File: host/tests/dsClasses/dsAudio_test_suite.yml [hunk 1]
```diff
@@ -22,7 +22,13 @@
...
     description: "dsAudio Device Settings testing profile / menu system for UT"
     test:
-        execute: "<PATH on Device>run.sh -p <PATH on Device>Sink_AudioSettings.yaml" #Execute command
+        artifacts:
+          - "../../../bin/hal_test"
+          - "../../../bin/libut_control.so"
+          - "../../../bin/run.sh"
+        execute:
+          command: "run.sh"
+          arguments: ""
```
Comment (Ulrond) [inline host/tests/dsClasses/dsAudio_test_suite.yml:31]: `arguments` should be removed.

File: host/tests/dsClasses/dsAudio_test_suite.yml [hunk 2]
```diff
@@ -22,7 +22,13 @@
 dsAudio:
     description: "dsAudio Device Settings testing profile / menu system for UT"
     test:
-        execute: "<PATH on Device>run.sh -p <PATH on Device>Sink_AudioSettings.yaml" #Execute command
+        artifacts:
+          - "../../../bin/hal_test"
```
Comment (Ulrond) [inline host/tests/dsClasses/dsAudio_test_suite.yml:26]: `should be a list of directories to copy`

**Review by Ulrond — CHANGES_REQUESTED** [review]
nearly

File: host/tests/dsClasses/dsAudio.py [hunk 3]
```diff
@@ -74,11 +75,7 @@ class dsAudioClass():
     This module provides common functionalities and extensions for the device Settings Audio Module.
     """
 
-    moduleName = "dsAudio"
-    menuConfig = os.path.join(dir_path, "dsAudio_test_suite.yml")
-    testSuite = "L3 dsAudio"
-
-    def __init__(self, deviceProfilePath:str, session=None ):
+    def __init__(self, deviceProfilePath:str, session=None, devicePath="/tmp"):
```
[the remark this answers is not in the document]
Comment (bhanucbp) [reply host/tests/dsClasses/dsAudio.py:78]: done

File: host/tests/dsClasses/dsAudio.py [hunk 4]
```diff
@@ -89,11 +86,20 @@ def __init__(self, deviceProfilePath:str, session=None ):
         Returns:
             None
         """
+        self.moduleName = "dsAudio"
+        self.menuConfig = os.path.join(dir_path, "dsAudio_test_suite.yml")
```
[the remark this answers is not in the document]
Comment (bhanucbp) [reply host/tests/dsClasses/dsAudio.py:90]: done

**Review by Ulrond — DISMISSED** [review]
only conversations to resolve

File: host/tests/dsClasses/dsAudio.py [hunk 5]
```diff
@@ -89,11 +86,20 @@ def __init__(self, deviceProfilePath:str, session=None ):
...
+        self.testSuite = "L3 dsAudio"
+
         # Load configurations for device profile and menu
         self.deviceProfile = ConfigRead( deviceProfilePath, self.moduleName)
         self.suitConfig    = ConfigRead(self.menuConfig, self.moduleName)
-        self.utMenu        = UTSuiteNavigatorClass(self.menuConfig, self.moduleName, session)
+        self.utMenu        = UTSuiteNavigatorClass(self.menuConfig, self.moduleName, session, devicePath)
```
[the remark this answers is not in the document]
Comment (Ulrond) [reply host/tests/dsClasses/dsAudio.py]: have updated this

File: host/tests/L3_TestCases/dsAudio/dsAudioHelperClass.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+        # List of streams with path
+        self.testStreams = []
+        url = []
+
+        streamPaths = self.testSetup.get("assets").get("device").get(self.testName).get("streams")
+
+        # Download test streams to device
+        if streamPaths and self.streamDownloadURL:
+            for s in streamPaths:
```
[the remark this answers is not in the document]
Comment (bhanucbp) [reply host/tests/L3_TestCases/dsAudio/dsAudioHelperClass.py]: changed the variable name

**Review by hari22yuva — APPROVED** [review]
looks ok

**Review by srinivasgtl — APPROVED** [review]
ship it.

**Pull request merged**